cmd/juju/model/trackbranch.go

```diff
@@ -78,21 +83,53 @@ func (c *trackBranchCommand) Info() *cmd.Info {
 // SetFlags implements part of the cmd.Command interface.
 func (c *trackBranchCommand) SetFlags(f *gnuflag.FlagSet) {
 	c.ModelCommandBase.SetFlags(f)
+	f.IntVar(&c.numUnits, "n", 0, "The number of units to track")
 }
 
 // Init implements part of the cmd.Command interface.
 func (c *trackBranchCommand) Init(args []string) error {
 	if len(args) == 0 {
 		return errors.Errorf("expected a branch name plus unit and/or application names(s)")
 	}
+	if c.numUnits < 0 {
```

Comment:
I think we should detect somehow if the user actually passed `-n 0` and return as invalid.
It would violate user expectation to pass that and have all units assigned.

Reply:
Ok, that makes sense.